internal/cloudproviders/azure/network.go

```diff
@@ -60,83 +60,83 @@ func (obj *AzureProvider) CreateVirtualNetwork(ctx context.Context, storage reso
 
 	if len(mainStateDocument.CloudInfra.Azure.VirtualNetworkName) != 0 {
 		log.Print("skipped virtualNetwork already created", "name", mainStateDocument.CloudInfra.Azure.VirtualNetworkName)
-		return nil
-	}
+	} else {
```

Comment:
do we actually need this
as the creation of VirtNet is a child in relation of `createNetwork()`

so there is no other resource creation dependent on this creation

as this else block is redundant and can be avoided